hefquin-engine/src/main/java/se/liu/ida/hefquin/engine/queryplan/utils/PhysicalPlanFactory.java

```diff
@@ -466,7 +466,7 @@ public static PhysicalPlan createPlan( final NaryLogicalOp rootOp,
 	public static PhysicalPlan createPlan( final NaryLogicalOp rootOp,
 	                                       final QueryPlanningInfo qpInfo,
 	                                       final List<PhysicalPlan> subplans ) {
-		final NaryPhysicalOp pop = LogicalToPhysicalOpConverter.convert(rootOp);
+		final NaryPhysicalOp pop = (NaryPhysicalOp) LogicalToPhysicalOpConverter.convert(rootOp);
```

Comment:
Also for this one, we need to push the `ExpectedVariables` object of every subplan into the `convert` function and, there, further into `PhysicalOpRegistry#create` and then into `PhysicalOpFactory#supports`.